it/src/test/java/apoc/it/core/SchemasEnterpriseFeaturesTest.java

```diff
@@ -164,6 +164,93 @@ private static void schemaNodeKeyAssertions(Map<String, Object> r) {
     }
 
 
+    @Test
+    public void testSchemaNodesWithNodeTypeConstraint() {
+        session.writeTransaction(tx -> {
+            tx.run("CREATE CONSTRAINT node_prop_type_movie FOR (m:Movie) REQUIRE (m.first) IS :: INTEGER");
+            tx.commit();
+            return null;
+        });
+
+        testResult(session, CALL_SCHEMA_NODES_ORDERED, (result) -> {
+            Map<String, Object> r = result.next();
+            assertEquals("Movie", r.get("label"));
+            assertEquals(List.of("first"), r.get("properties"));
+            assertEquals(":Movie(first)", r.get("name"));
+            assertEquals("", r.get("status"));
+            assertEquals("NODE_PROPERTY_TYPE", r.get("type"));
+            final String expectedUserDescConstraint = "name='node_prop_type_movie', type='NODE PROPERTY TYPE', schema=(:Movie {first}), propertyType=INTEGER";
+            Assertions.assertThat(r.get("userDescription").toString()).contains(expectedUserDescConstraint);
+
+            assertFalse(result.hasNext());
+        });
+    }
+
+    @Test
+    public void testNodeTypeConstraintIsKeptAndDropped() {
+        session.writeTransaction(tx -> {
+            tx.run("CREATE CONSTRAINT node_prop_type_movie_first FOR (m:Movie) REQUIRE (m.first) IS :: INTEGER");
+            tx.run("CREATE CONSTRAINT node_prop_type_movie_second FOR (m:Movie) REQUIRE (m.second) IS :: INTEGER");
+            tx.commit();
+            return null;
+        });
+
+        HashSet<AssertSchemaResult> expectedResult = new HashSet<>();
+        AssertSchemaResult firstConstraint = new AssertSchemaResult("Movie", List.of("first"));
+        firstConstraint.unique = false;
+        expectedResult.add(firstConstraint);
+
+        AssertSchemaResult secondConstraint = new AssertSchemaResult("Movie", List.of("second"));
+        secondConstraint.unique = false;
+        secondConstraint.action = "DROPPED";
+        expectedResult.add(secondConstraint);
+
+        HashSet<AssertSchemaResult> actualResult = new HashSet<>();
+
+
+        testResult(session, "CALL apoc.schema.assert({},{Movie:[[\"first\"]]})", (result) -> {
+            while (result.hasNext()) {
+                Map<String, Object> r = result.next();
+                AssertSchemaResult con = new AssertSchemaResult(r.get("label"), (List<String>) r.get("keys"));
+                con.unique = (boolean) r.get("unique");
+                con.action = (String) r.get("action");
+                actualResult.add(con);
+
+                assertEquals(1, expectedResult.stream().filter(
+                        c -> c.keys.containsAll(con.keys) &&
+                                c.action.equals(con.action) &&
+                                c.label.equals(con.label) &&
+                                c.unique == con.unique
+                ).toList().size());
+            }
+
+            assertEquals(expectedResult.size(), actualResult.size());
+        });
```

Comment:
Could you explain me what we are trying to test here 🙏 ?

Reply:
So I copied the style for each type of test as I figured being consistent per test type was less confusing than per constraint type. But what happens with this one is that above the loop is written the 2 expected results, and this checks that both results exist in the result set. The first constraint is in the constraint list of the CALL so should be kept, the second constraint is not mentioned so should be dropped. So the test checks one is kept and one is dropped :)